api/apps/api/src/modules/projects/job-status/job-status.view.api.entity.ts

```diff
@@ -0,0 +1,76 @@
+import { ViewColumn, ViewEntity } from 'typeorm';
+import { JobType } from '@marxan-api/modules/projects/job-status/jobs.enum';
+import { JobStatus } from '@marxan-api/modules/scenarios/scenario.api.entity';
+import { API_EVENT_KINDS } from '@marxan/api-events';
+
+@ViewEntity({
+  expression: `
+    SELECT
+      DISTINCT ON ("jobType", topic) "jobType",
+      api_events.topic AS "scenarioId",
+      projects.id AS "projectId",
+      api_events.kind
+    FROM
+      api_events
+      INNER JOIN scenarios ON api_events.topic = scenarios.id
+      INNER JOIN projects ON projects.id = scenarios.project_id
+      CROSS JOIN LATERAL SUBSTRING(
+        api_events.kind
+        FROM
+          'scenario.#"[^.]*#"%' FOR '#'
+      ) AS "jobType"
+    ORDER BY
+      "jobType",
+      api_events.topic,
+      api_events.timestamp DESC;
+  `,
+})
+export class ScenarioJobStatus {
+  @ViewColumn()
+  jobType!: JobType;
```

Comment:
It would be best to have it defined as a naming strategy of the TypeORM, as it's the main user of most of the tables, and then we wouldn't need to name every column by hand.
I'm changing it anyway.

Reply:
fixed in 68d74ba1

Reply:
thanks - you are right about TypeORM being the main user of most of the tables. Personally I prefer the added verbosity of using `name` parameters to decorators over the added care needed to quote identifiers in raw SQL or other SQL queries outside of the app codebase (e.g raw SQL in Python notebooks, and so on) - given that we've used snake_case so far, let's stick to it.